next.config.js

```diff
@@ -1,16 +1,29 @@
 /** @type {import('next').NextConfig} */
 module.exports = {
-  output: 'standalone',
+  output: "standalone",
   images: {
-    domains: [
-      'play.google.com',
-      'res.cloudinary.com'],
-  }
+    domains: ["play.google.com", "res.cloudinary.com"],
+  },
+  publicRuntimeConfig: {
+    BASE_URL: process.env.BASE_URL || "https://devme.bettersocial.org",
+    DOWNLOAD_BETTERSOCIAL_APP_URL:
+      process.env.DOWNLOAD_BETTERSOCIAL_APP_URL ||
+      "https://linkdev.bettersocial.org/download",
+    HUMAN_INTERNET_URL:
+      process.env.HUMAN_INTERNET_URL || "https://human-internet.org/",
+    DYNAMIC_LINK_DOMAIN:
+      process.env.DYNAMIC_LINK_DOMAIN || "https://devme.bettersocial.org",
+  },
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The configuration values are directly read from environment variables with fallbacks to hardcoded URLs. This could lead to unexpected behavior if the environment variables are not set correctly. It would be better to validate these environment variables during startup and fail fast if they are not set. This way, you can avoid runtime errors due to misconfiguration.

```diff
+ const requiredConfig = ['BASE_URL', 'DOWNLOAD_BETTERSOCIAL_APP_URL', 'HUMAN_INTERNET_URL', 'DYNAMIC_LINK_DOMAIN'];
+ 
+ requiredConfig.forEach((config) => {
+   if (!process.env[config]) {
+     throw new Error(`Environment variable ${config} is not set`);
+   }
+ });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->